src/main/java/org/folio/circulation/domain/notice/schedule/RequestScheduledNoticeHandler.java

```diff
@@ -61,9 +68,33 @@ private RequestScheduledNoticeHandler(RequestRepository requestRepository,
 
   private CompletableFuture<Result<ScheduledNotice>> handleRequestNotice(ScheduledNotice notice) {
     return requestRepository.getById(notice.getRequestId())
-      .thenApply(r -> r.map(RequestAndRelatedRecords::new))
-      .thenCompose(r -> r.after(records -> sendNotice(records, notice)))
-      .thenCompose(r -> r.after(records -> updateNotice(records, notice)));
+      .thenCompose(r -> r.afterWhen(
+        request -> noticeShouldBeDeleted(request, notice),
```

Comment:
```suggestion
        request -> noticeIsIrrelevant(request, notice),
```

Notice should also be deleted (after sending) if it is relevant, but not recurring. So the method name `noticeShouldBeDeleted` can be a bit misleading.